astropy/units/quantity_helper/function_helpers.py

```diff
@@ -109,7 +114,7 @@
         np.product,  # noqa: NPY003, NPY201
         np.cumproduct,  # noqa: NPY003, NPY201
     }
-else:
+if not NUMPY_LT_2_0:
```

Comment:
Can't this remain an `else`?

Reply:
It could, but it's bound to take this form sooner or later, when numpy<2 is dropped. I also find that if/else becomes more confusing as soon as there are more than 2 cases to deal with. If you're not convinced, please feel free to push to this branch, as I won't be able to until tomorrow morning.

Reply:
Wait I was wrong. It doesn't need to evolve as I described, we'll just deindent it when it's not conditional anymore. My second point still stands though.

Reply:
The second `if` (for ` not NUMPY_LT_2_1`) shouldn't be changed. So this will remain a 2-case if/else. 

```python
if NUMPY_LT_2_0: ...
else: ...
if not NUMPY_LT_2_1: ...
```

Which in a few years will become

```python
...
if not NUMPY_LT_2_1: ...
```

Reply:
Triple `if` is still slightly more appealing to me but I don't hold strong opinions here.

Reply:
I'm with @nstarman here - prefer the `else`, but so slightly that one might as well not run CI for it again.